### PR TITLE
Suggest fixes

### PR DESCRIPTION
**Pull request by pwalczysko**
Adjusting the playbook to the tested working version

1. Adjust the hosts-all which did not work in testing
2. Adjust the postgresql version (this passes on ome-training-3 in checkmode)

An open question here is that the setup was not tested on a clean CentOS 7 machine - how to create one easily please cc @joshmoore @sbesson ? I tested only on training-3 in check mode.

Tested setup
``ansible-playbook --become -i hosts.yml playbook.yml`` 
was working when the host in ``hosts.yml`` was replaced with training-3.

See also
https://github.com/ome/omero-guides/pull/232

### CODE REVIEW

File: hosts.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+all:
```
Comment (sbesson) [inline hosts.yml:1]:
```suggestion
---
all:
```

**Review by sbesson — CHANGES_REQUESTED** [review]
One syntactic change to fix Ansible lint (reactivated as the Molecule workflow had been disabled)

File: playbook.yml [hunk 2]
```diff
@@ -18,4 +18,4 @@
       omero_server_rootpassword: ChangeMe
     - role: ome.omero_web
   vars:
-    postgresql_version: "11"
+    postgresql_version: "13"
```
Comment (sbesson) [inline playbook.yml:21]:
```
     E       assert False
    E        +  where False = <service postgresql-11>.is_running
```

The Molecule tests will need to be adjusted to test the `postgresql-13` service

**Review by sbesson — CHANGES_REQUESTED** [review]

**Review by sbesson — APPROVED** [review]

**Comment by pwalczysko** [comment]
@sbesson I have created a clean docker with CentOS 7 and ran the playbook after installing the requirements.

I got

```
[root@388b5a8730ad ~]# ansible-playbook --connection=local -i hosts.yml  playbook.yml
...


TASK [ome.postgresql_client : postgres | setup yum repository] *******************
fatal: [localhost]: FAILED! => {"changed": false, "msg": "Failure downloading https://download.postgresql.org/pub/repos/yum/reporpms/EL-7-x86_64/pgdg-redhat-repo-latest.noarch.rpm, Request failed: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:579)>"}
```

I checked the version of the ome.postgresql role and it is 5.2.0 which matches what we have in prod-playbooks.

**Comment by sbesson** [comment]
Have you ran either `yum update *` or minimally `yum update ca-certificates` on the original Docker image?

**Comment by pwalczysko** [comment]
Thanks, the ``yum update ca-certificates`` solved the problem, the task completed.

But now I am having

```
TASK [ome.postgresql_client : postgres | install client packages] ****************
fatal: [localhost]: FAILED! => {"changed": false, "msg": "No package matching 'postgresql13' found available, installed or updated", "rc": 126, "results": ["No package matching 'postgresql13' found available, installed or updated"]}
```

**Comment by sbesson** [comment]
What's the version of `ome.postgresql_client`?

**Comment by pwalczysko** [comment]
> What's the version of `ome.postgresql_client`? I think 0.2.0 is required for PSQL 13 support

You mean what does the line in the ``requirements.yml`` in this repo which corresponds to [- src: ome.postgresql_client](https://github.com/ome/prod-playbooks/blob/91ef217d2fb4f8be0f5697bb3ca0b057124b98b2/requirements.yml#L70) say ?
There is no such corresponding line in the https://github.com/ome/ansible-example-omero-onenode/blob/master/requirements.yml - maybe it is missing ?

**Comment by pwalczysko** [comment]
Indeed, when I added the https://github.com/ome/prod-playbooks/blob/91ef217d2fb4f8be0f5697bb3ca0b057124b98b2/requirements.yml#L70 to the [requirements](https://github.com/ome/ansible-example-omero-onenode/blob/master/requirements.yml), and rerun the playbook, many many tasks finished successfully, so it has helped and probably
the line should be added ? I can open a new PR. But after this, the next failure is

```
TASK [ome.postgresql : postgres | initialise PostgreSQL cluster (skip if data directory already exists)] ***
fatal: [localhost]: FAILED! => {"changed": true, "cmd": ["/usr/pgsql-13/bin/postgresql-13-setup", "initdb"], "delta": "0:00:00.251592", "end": "2022-08-10 15:33:11.980763", "msg": "non-zero return code", "rc": 1, "start": "2022-08-10 15:33:11.729171", "stderr": "Failed to get D-Bus connection: Operation not permitted", "stderr_lines": ["Failed to get D-Bus connection: Operation not permitted"], "stdout": "failed to find PGDATA setting in postgresql-13.service", "stdout_lines": ["failed to find PGDATA setting in postgresql-13.service"]}
```

**Comment by sbesson** [comment]
> "Failed to get D-Bus connection: Operation not permitted"

I think this error points at an issue with the initialisation of systemd on CentOS 7. This is the reason why the Molecule tests are not running against the vanilla `centos:7` image but instead `centos:systemd` with `/sbin/init` as the command - see [/sbin/init](https://github.com/ome/ansible-example-omero-onenode/blob/43f1231d9bb92670dd8ece0b768f6e36aad61b46/molecule/centos7/molecule.yml#L11-L15)

**Comment by pwalczysko** [comment]
> > "Failed to get D-Bus connection: Operation not permitted"
> 
> I think this error points at an issue with the initialisation of systemd on CentOS 7. This is the reason why the Molecule tests are not running against the vanilla `centos:7` image but instead `centos:systemd` with `/sbin/init` as the command - see [/sbin/init](https://github.com/ome/ansible-example-omero-onenode/blob/43f1231d9bb92670dd8ece0b768f6e36aad61b46/molecule/centos7/molecule.yml#L11-L15)

Thank you, this helped a lot. I took an image with systemd creating it from a Dockerfile which looked like

```
FROM centos/systemd

MAINTAINER "Your Name" <you@example.com>

CMD ["/sbin/init"]
```

Then I built the docker and ran it and 

```
docker build -f Dockerfile.j2 --rm --no-cache -t httpd2 . 
docker run --privileged --name httpd2 -v /sys/fs/cgroup:/sys/fs/cgroup:ro  -d  httpd
docker exec -it httpd2 bash
```

I had to do all the things you recommended, and also some more

```
yum update ca-certificates
yum install sudo
```

The hardest bit was to install pip to be able to install ansible. The python 3.6 had to be installed first, then pip.
 
After some trials, the playbook ran fine to the end, I am just realizing I am not able to really check OMERO.web, as my docker has no port forwarding (?).

But I am able to log in to OMERO and the service for web is claiming it is up

<details>

```
(venv3) bash-4.2$ omero login
Previous session expired for root on localhost:4064
Server: [localhost:4064]
Username: [root]

Password:
Created session for root@localhost:4064. Idle timeout: 10 min. Current group: system

...
[root@2f2c6845ecb2 /]# systemctl status omero-web.service -l
● omero-web.service - OMERO.web
   Loaded: loaded (/etc/systemd/system/omero-web.service; enabled; vendor preset: disabled)
   Active: active (running) since Thu 2022-08-11 11:53:41 UTC; 2min 23s ago
 Main PID: 8401 (gunicorn)
   CGroup: /docker/2f2c6845ecb2707ba337328a2c4097b4f7889951774d16ad6f6f525e82ffe123/system.slice/omero-web.service
           ├─8401 /opt/omero/web/venv3/bin/python3 /opt/omero/web/venv3/bin/gunicorn -D -p /opt/omero/web/OMERO.web/var/django.pid --bind 127.0.0.1:4080 --workers 5 --threads 1 --timeout 60 --max-requests 0 omeroweb.wsgi:application
           ├─8404 /opt/omero/web/venv3/bin/python3 /opt/omero/web/venv3/bin/gunicorn -D -p /opt/omero/web/OMERO.web/var/django.pid --bind 127.0.0.1:4080 --workers 5 --threads 1 --timeout 60 --max-requests 0 omeroweb.wsgi:application
           ├─8410 /opt/omero/web/venv3/bin/python3 /opt/omero/web/venv3/bin/gunicorn -D -p /opt/omero/web/OMERO.web/var/django.pid --bind 127.0.0.1:4080 --workers 5 --threads 1 --timeout 60 --max-requests 0 omeroweb.wsgi:application
           ├─8411 /opt/omero/web/venv3/bin/python3 /opt/omero/web/venv3/bin/gunicorn -D -p /opt/omero/web/OMERO.web/var/django.pid --bind 127.0.0.1:4080 --workers 5 --threads 1 --timeout 60 --max-requests 0 omeroweb.wsgi:application
           ├─8413 /opt/omero/web/venv3/bin/python3 /opt/omero/web/venv3/bin/gunicorn -D -p /opt/omero/web/OMERO.web/var/django.pid --bind 127.0.0.1:4080 --workers 5 --threads 1 --timeout 60 --max-requests 0 omeroweb.wsgi:application
           └─8415 /opt/omero/web/venv3/bin/python3 /opt/omero/web/venv3/bin/gunicorn -D -p /opt/omero/web/OMERO.web/var/django.pid --bind 127.0.0.1:4080 --workers 5 --threads 1 --timeout 60 --max-requests 0 omeroweb.wsgi:application

Aug 11 11:53:25 2f2c6845ecb2 systemd[1]: Starting OMERO.web...
Aug 11 11:53:35 2f2c6845ecb2 omero[8379]: 0 static files copied to '/opt/omero/web/OMERO.web/var/static', 530 unmodified, 2 post-processed.
Aug 11 11:53:41 2f2c6845ecb2 omero[8379]: Clearing expired sessions. This may take some time... [OK]
Aug 11 11:53:41 2f2c6845ecb2 omero[8379]: Starting OMERO.web... [OK]
Aug 11 11:53:41 2f2c6845ecb2 systemd[1]: Can't open PID file /opt/omero/web/OMERO.web/var/django.pid (yet?) after start: No such file or directory
Aug 11 11:53:41 2f2c6845ecb2 systemd[1]: Started OMERO.web.
```

</details>

**Comment by joshmoore** [comment]
Did this discussion migrate elsewhere?

**Comment by pwalczysko** [comment]
Yes, the main outstanding issue is captured in https://github.com/ome/ansible-example-omero-onenode/issues/13 - the other issues are resolved

**Pull request merged**